rskj-core/src/main/java/co/rsk/trie/Trie.java

```diff
@@ -563,7 +571,13 @@ public Trie delete(String key) {
             buffer.put(this.getValue());
         }
 
-        return buffer.array();
+        encoded = buffer.array();
+
+        return cloneArray(encoded);
+    }
+
+    public int getMessageLength() {
+        return toMessage().length;
```

Comment:
unnecessary clone here.

Reply:
We made this change to address this clone https://github.com/rsksmart/rskj/pull/874